controllers/osccluster_routetable_controller_unit_test.go

```diff
@@ -0,0 +1,3292 @@
+		expCheckRouteTableOscDuplicateNameErr error
+	}{
+		{
+			name:                                  "get no duplicate routeTable Name",
+			spec:                                  defaultRouteTableGatewayInitialize,
+			expCheckRouteTableOscDuplicateNameErr: nil,
+		},
+		{
+			name: "get duplicate routeTable Name",
+			spec: infrastructurev1beta1.OscClusterSpec{
+				Network: infrastructurev1beta1.OscNetwork{
+					Net: infrastructurev1beta1.OscNet{
+						Name:    "test-net",
+						IpRange: "10.0.0.0/16",
+					},
+					Subnets: []*infrastructurev1beta1.OscSubnet{
+						{
+							Name:          "test-subnet",
+							IpSubnetRange: "10.0.0.0/24",
+						},
+					},
+					InternetService: infrastructurev1beta1.OscInternetService{
+						Name: "test-internetservice",
+					},
+					RouteTables: []*infrastructurev1beta1.OscRouteTable{
+						{
+							Name:       "test-routetable",
+							SubnetName: "test-subnet",
+							Routes: []infrastructurev1beta1.OscRoute{
+								{
+									Name:        "test-route",
+									TargetName:  "test-internetservice",
+									TargetType:  "gateway",
+									Destination: "0.0.0.0/0",
+								},
+							},
+						},
+						{
+							Name:       "test-routetable",
+							SubnetName: "test-subnet",
+							Routes: []infrastructurev1beta1.OscRoute{
+								{
+									Name:        "test-route",
+									TargetName:  "test-internetservice",
+									TargetType:  "gateway",
+									Destination: "0.0.0.0/0",
+								},
+							},
+						},
+					},
+				},
+			},
+			expCheckRouteTableOscDuplicateNameErr: fmt.Errorf("test-routetable already exist"),
+		},
+	}
+	for _, rttc := range routeTableTestCases {
+		t.Run(rttc.name, func(t *testing.T) {
+			t.Logf("Validate to %s", rttc.name)
+			log := klogr.New()
+			oscCluster := infrastructurev1beta1.OscCluster{
+				Spec: rttc.spec,
+				ObjectMeta: metav1.ObjectMeta{
+					UID: "uid",
+				},
+			}
+			clusterScope := &scope.ClusterScope{
+				Logger: log,
+				Cluster: &clusterv1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{
+						UID: "uid",
+					},
+				},
+				OscCluster: &oscCluster,
+			}
+			duplicateResourceRouteTableNameErr := checkRouteTableOscDuplicateName(clusterScope)
+			if duplicateResourceRouteTableNameErr != nil {
+				if duplicateResourceRouteTableNameErr.Error() != rttc.expCheckRouteTableOscDuplicateNameErr.Error() {
+					t.Errorf("checkRouteTableOscDuplicateName() expected error = %s, received error %s", rttc.expCheckRouteTableOscDuplicateNameErr, duplicateResourceRouteTableNameErr.Error())
+				}
+			}
+		})
+	}
+}
+
+// TestCheckRouteOscDuplicateName has several tests to cover the code of the func checkRouteOscDuplicateName
+func TestCheckRouteOscDuplicateName(t *testing.T) {
+	routeTestCases := []struct {
+		name                             string
+		spec                             infrastructurev1beta1.OscClusterSpec
+		expCheckRouteOscDuplicateNameErr error
+	}{
+		{
+			name: "check default value",
```

Comment:
With this test case name, I expect that we check in the test that an empty configuration will take default value. The test here only check that the function does not throw an error

Reply:
i put a better explicate name